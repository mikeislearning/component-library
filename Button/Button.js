// @format
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './Button.styles';

export default class Button extends React.Component {
  static defaultProps = {
    disabled: false,
    fullWidth: false,
    theme: 'primary',
  };

  static propTypes = {
    accessibilityLabel: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
    disabled: PropTypes.bool,
    fullWidth: PropTypes.bool,
    onPress: PropTypes.func,
    textStyle: PropTypes.object,
    theme: PropTypes.string,
  };

  render() {
    const {
      accessibilityLabel,
      children,
      disabled,
      fullWidth,
      onPress,
      textStyle,
      theme,
    } = this.props;
    const buttonStyles = [
      styles.button,
      styles[theme],
      disabled ? styles.disabled : null,
      fullWidth ? styles.fullWidth : null,
    ];
    const textStyles = [styles.text, styles[`${theme}Text`]];
    if (textStyle) textStyles.push(textStyle);
    if (disabled) textStyles.push(styles.disabledText);

    return (
      <TouchableOpacity
        accessibilityLabel={accessibilityLabel}
        disabled={disabled}
        onPress={onPress}
        style={buttonStyles}
      >
        {typeof children === 'string' ? <Text style={textStyles}>{children}</Text> : children}
      </TouchableOpacity>
    );
  }
}
