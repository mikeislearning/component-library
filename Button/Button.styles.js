import { StyleSheet } from 'react-native';

import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  bordered: {
    backgroundColor: colors.white,
    borderColor: colors.blue,
    borderWidth: 1,
    padding: 14,
  },
  borderedText: {
    color: colors.blue,
    textAlign: 'center',
  },
  button: {
    alignItems: 'center',
    borderRadius: 6,
    justifyContent: 'center',
  },
  disabled: {
    backgroundColor: colors.gray.light,
    borderColor: colors.gray.light,
  },
  disabledText: {
    color: colors.gray.medium,
  },
  fullWidth: {
    width: '100%',
  },
  primary: {
    backgroundColor: colors.blue,
    borderColor: colors.blue,
    borderWidth: 1,
    padding: 15,
  },
  primaryText: {
    color: colors.white,
    textAlign: 'center',
  },
  quaternary: {
    backgroundColor: colors.white,
  },
  quaternaryText: {
    color: colors.purple,
    fontSize: 17,
    textAlign: 'center',
  },
  secondary: {
    backgroundColor: colors.white,
    borderColor: colors.gray.light,
    borderWidth: 1,
    padding: 15,
  },
  secondaryText: {
    color: colors.black,
  },
  tertiary: {
    backgroundColor: colors.gray.light,
    borderColor: colors.gray.light,
    borderWidth: 1,
    padding: 8,
  },
  tertiaryText: {
    color: colors.black,
  },
  text: {
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
  },
  transparent: {
    backgroundColor: colors.transparent,
    padding: 12,
  },
  transparentText: {
    color: colors.gray.medium,
  },
});

export default styles;
