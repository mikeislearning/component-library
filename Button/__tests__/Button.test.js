/**
 * @format
 */

// import 'react-native';
import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import Button from '../Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const button = render(<Button />);
    expect(button).toMatchSnapshot();
  });

  it('can be pressed successfully', () => {
    const mockedFn = jest.fn();
    const { getByText } = render(<Button onPress={mockedFn}>Testing</Button>);

    fireEvent.press(getByText('Testing'));

    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
