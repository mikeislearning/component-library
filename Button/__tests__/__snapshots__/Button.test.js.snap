// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4145C8",
      "borderColor": "#4145C8",
      "borderRadius": 6,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
    }
  }
/>
`;
